src/couch_scanner/src/couch_scanner_plugin_conflict_finder.erl

```diff
@@ -0,0 +1,171 @@
+
+-define(CONFLICTS, <<"conflicts">>).
+-define(DELETED_CONFLICTS, <<"deleted_conflicts">>).
+
+-define(OPTS, #{
+    ?CONFLICTS => true,
+    ?DELETED_CONFLICTS => true
+}).
+
+% Behavior callbacks
+
+start(SId, #{}) ->
+    St = init_config(#st{sid = SId}),
+    ?INFO("Starting.", [], #{sid => SId}),
+    {ok, St}.
+
+resume(SId, #{<<"opts">> := OldOpts}) ->
+    St = init_config(#st{sid = SId}),
+    case OldOpts == St#st.opts of
+        true ->
+            ?INFO("Resuming.", [], #{sid => SId}),
+            {ok, St};
+        false ->
+            ?INFO("Resetting. Config changed.", [], #{sid => SId}),
+            reset
+    end.
+
+complete(#st{sid = SId, dbname = DbName, report = Report} = St) ->
+    report_per_db(St, DbName, Report),
+    ?INFO("Completed", [], #{sid => SId}),
+    {ok, #{}}.
+
+checkpoint(#st{sid = SId, opts = Opts}) ->
+    case Opts == opts() of
+        true ->
+            {ok, #{<<"opts">> => Opts}};
+        false ->
+            ?INFO("Resetting. Config changed.", [], #{sid => SId}),
+            reset
+    end.
+
+db(#st{} = St, _DbName) ->
+    {ok, St}.
+
+doc_id(#st{} = St, <<?DESIGN_DOC_PREFIX, _/binary>>, _Db) ->
+    {skip, St};
+doc_id(#st{} = St, DocId, Db) ->
+    {ok, #doc_info{revs = Revs}} = couch_db:get_doc_info(Db, DocId),
+    DbName = mem3:dbname(couch_db:name(Db)),
+    {ok, check(St, DbName, DocId, Revs)}.
+
+% Private
+
+init_config(#st{} = St) ->
+    St#st{
+        opts = opts(),
+        doc_report = cfg_bool("doc_report", St#st.doc_report)
+    }.
+
+check(#st{} = St, _, _, Revs) when length(Revs) =< 1 ->
+    St;
+check(#st{doc_report = true, opts = Opts} = St, DbName, DocId, Revs) ->
+    {DeletedConflicts, Conflicts} =
+        lists:partition(fun(R) -> R#rev_info.deleted end, Revs),
+    ConflictsReport = gen_report(doc, ?CONFLICTS, Opts, Conflicts),
+    DeletedConflictsReport = gen_report(doc, ?DELETED_CONFLICTS, Opts, DeletedConflicts),
+    DocReport = maps:merge(ConflictsReport, DeletedConflictsReport),
+    report_per_doc(#st{} = St, DbName, DocId, DocReport),
+    DbReport = maps:from_list([{K, tuple_size(V)} || {K, V} <- maps:to_list(DocReport)]),
+    report(#st{} = St, DbName, DbReport);
+check(#st{opts = Opts} = St, DbName, _DocId, Revs) ->
+    {DeletedConflicts, Conflicts} =
+        lists:partition(fun(R) -> R#rev_info.deleted end, Revs),
+    ConflictsReport = gen_report(db, ?CONFLICTS, Opts, Conflicts),
+    DeletedConflictsReport = gen_report(db, ?DELETED_CONFLICTS, Opts, DeletedConflicts),
+    DbReport = maps:merge(ConflictsReport, DeletedConflictsReport),
+    report(#st{} = St, DbName, DbReport).
+
+gen_report(db, ?CONFLICTS, #{?CONFLICTS := true}, Revs) ->
+    #{?CONFLICTS => length(Revs)};
+gen_report(db, ?DELETED_CONFLICTS, #{?DELETED_CONFLICTS := true}, Revs) ->
+    #{?DELETED_CONFLICTS => length(Revs)};
+gen_report(doc, ?CONFLICTS, #{?CONFLICTS := true}, Revs) ->
+    #{?CONFLICTS => list_to_tuple([couch_doc:rev_to_str(R#rev_info.rev) || R <- Revs])};
```

Comment:
In general it is not a good idea to use tuples for variable length items.  Yes, they are more memory efficient and allows element-wise O(1) access but unless we absolutely need to do that it maybe better to stick with a list

Reply:
Convert back to list and add a `flatten_one_level/1` function to print out the report properly.